src/appshell/view/appmenumodel.cpp

```diff
@@ -72,6 +75,30 @@ void AppMenuModel::load()
     appMenuModelHook()->onAppMenuInited();
 }
 
+bool AppMenuModel::isGlobalMenuAvailable()
+{
+#if defined(Q_OS_MACOS)
+    return true;
+#elif defined(Q_OS_LINUX)
+    return isDBusGlobalMenuAvailable();
+#else
+    return false;
+#endif
```

Comment:
I think better move this method to `IPlatformTheme` interface and use it here

Reply:
It would be better to keep the `isGlobalMenuAvailable()` method in `AppMenuModel` (because it's already exposed to QML) and add platform-specific methods to subclasses of `IPlatformTheme`

Reply:
Yes, exactly